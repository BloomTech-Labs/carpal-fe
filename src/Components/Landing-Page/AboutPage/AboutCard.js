import React from "react";
import ImagesSocials from "./ImagesSocials";

const AboutCard = (props) => {
    return (
        <div className="awesome-aliens-box" data-testid="awesome-aliens-box">
            <div className="left" data-testid="left">
                <h2 data-testid="carpaller-name">{props.carpaller.name}</h2>
                <h3 className="team-title">{props.carpaller.title}</h3>
                <div className="hide-tablet show-desktop">
                    <ImagesSocials carpaller={props.carpaller} />
                </div>

                <p>{props.carpaller.bio}</p>
            </div>
            <div className="right show-tablet hide-desktop" data-testid="right">
                <ImagesSocials carpaller={props.carpaller} />
            </div>
        </div>
    );
};

export default AboutCard;
