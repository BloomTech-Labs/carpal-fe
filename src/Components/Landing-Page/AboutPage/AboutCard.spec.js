import React from "react";
import * as rtl from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { BrowserRouter as Router } from "react-router-dom";
import AboutCard from "./AboutCard";

afterEach(rtl.cleanup);

let props = {
    carpaller: {
        name: "Dennis Padiernos",
        title: "Chief Dad Joke Officer",
        cutieImg: "blueCutie.png",
        profileImg: "dennis.png",
        githubURL: "https://github.com/depadiernos",
        linkedinURL: "https://www.linkedin.com/in/dennispadiernos",
        bio:
            "Dennis is a software engineer who also loves music and mechanical keyboards and in his free time, he likes long walks by the beach and being a secret agent of OWCA (Organization Without a Cool Acronym)"
    }
};

describe("about card renders without crashing", () => {
    test("<AboutCard /> snapshot", () => {
        const wrapper = rtl.render(
            <Router>
                <AboutCard {...props} />
            </Router>
        );
        expect(wrapper).toMatchSnapshot();
    });
    test("structure displays properly", () => {
        const { getByTestId } = rtl.render(
            <Router>
                <AboutCard {...props} />
            </Router>
        );
        expect(getByTestId("awesome-aliens-box")).toBeVisible();
        expect(getByTestId("left")).toBeVisible();
        expect(getByTestId("right")).toBeVisible();
        expect(getByTestId("carpaller-name")).toHaveTextContent(
            /Dennis Padiernos/
        );
    });
});
