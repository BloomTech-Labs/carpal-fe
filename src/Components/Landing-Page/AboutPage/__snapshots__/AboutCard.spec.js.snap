// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`about card renders without crashing <AboutCard /> snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="awesome-aliens-box"
        data-testid="awesome-aliens-box"
      >
        <div
          class="left"
          data-testid="left"
        >
          <h2
            data-testid="carpaller-name"
          >
            Dennis Padiernos
          </h2>
          <h3
            class="team-title"
          >
            Chief Dad Joke Officer
          </h3>
          <div
            class="hide-tablet show-desktop"
          >
            <div
              class="img-container"
              data-testid="img-container"
            >
              <img
                alt="CarPal cutie"
                class="over"
                src="blueCutie.png"
              />
              <img
                alt="Dennis Padiernos"
                class="under"
                src="dennis.png"
              />
            </div>
            <div
              class="socials"
              data-testid="socials"
            >
              <a
                data-testid="github-url"
                href="https://github.com/depadiernos"
                target="_blank"
              >
                <i
                  class="fab fa-github fa-3x"
                />
              </a>
              <a
                href="https://www.linkedin.com/in/dennispadiernos"
                target="_blank"
              >
                <i
                  class="fab fa-linkedin-in fa-3x"
                />
              </a>
            </div>
          </div>
          <p>
            Dennis is a software engineer who also loves music and mechanical keyboards and in his free time, he likes long walks by the beach and being a secret agent of OWCA (Organization Without a Cool Acronym)
          </p>
        </div>
        <div
          class="right show-tablet hide-desktop"
          data-testid="right"
        >
          <div
            class="img-container"
            data-testid="img-container"
          >
            <img
              alt="CarPal cutie"
              class="over"
              src="blueCutie.png"
            />
            <img
              alt="Dennis Padiernos"
              class="under"
              src="dennis.png"
            />
          </div>
          <div
            class="socials"
            data-testid="socials"
          >
            <a
              data-testid="github-url"
              href="https://github.com/depadiernos"
              target="_blank"
            >
              <i
                class="fab fa-github fa-3x"
              />
            </a>
            <a
              href="https://www.linkedin.com/in/dennispadiernos"
              target="_blank"
            >
              <i
                class="fab fa-linkedin-in fa-3x"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="awesome-aliens-box"
      data-testid="awesome-aliens-box"
    >
      <div
        class="left"
        data-testid="left"
      >
        <h2
          data-testid="carpaller-name"
        >
          Dennis Padiernos
        </h2>
        <h3
          class="team-title"
        >
          Chief Dad Joke Officer
        </h3>
        <div
          class="hide-tablet show-desktop"
        >
          <div
            class="img-container"
            data-testid="img-container"
          >
            <img
              alt="CarPal cutie"
              class="over"
              src="blueCutie.png"
            />
            <img
              alt="Dennis Padiernos"
              class="under"
              src="dennis.png"
            />
          </div>
          <div
            class="socials"
            data-testid="socials"
          >
            <a
              data-testid="github-url"
              href="https://github.com/depadiernos"
              target="_blank"
            >
              <i
                class="fab fa-github fa-3x"
              />
            </a>
            <a
              href="https://www.linkedin.com/in/dennispadiernos"
              target="_blank"
            >
              <i
                class="fab fa-linkedin-in fa-3x"
              />
            </a>
          </div>
        </div>
        <p>
          Dennis is a software engineer who also loves music and mechanical keyboards and in his free time, he likes long walks by the beach and being a secret agent of OWCA (Organization Without a Cool Acronym)
        </p>
      </div>
      <div
        class="right show-tablet hide-desktop"
        data-testid="right"
      >
        <div
          class="img-container"
          data-testid="img-container"
        >
          <img
            alt="CarPal cutie"
            class="over"
            src="blueCutie.png"
          />
          <img
            alt="Dennis Padiernos"
            class="under"
            src="dennis.png"
          />
        </div>
        <div
          class="socials"
          data-testid="socials"
        >
          <a
            data-testid="github-url"
            href="https://github.com/depadiernos"
            target="_blank"
          >
            <i
              class="fab fa-github fa-3x"
            />
          </a>
          <a
            href="https://www.linkedin.com/in/dennispadiernos"
            target="_blank"
          >
            <i
              class="fab fa-linkedin-in fa-3x"
            />
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
